### ISSUE TITLE
IAssemblyProvider

### ISSUE THREAD

**Issue by fraserkyle** [ExtCore/ExtCore#105]
I beleive that the IAssemblyProvider interface is too specific.  I intend to use ExtCore to create a simple pluggable CMS which will allow installation of Packages containing ExtCore extensions.  

Packages will be created as projects, containing json files, detailing the content of each package.  A database table will be maintained containing reference to installed packages.  My intention is to create a new Assembly Provider which will return assemblies for the packages recorded as installed on my SQL table.

Is this the intended use for the IAssemblyProvider interface?  If so then can the path and includeSubpaths parameters within GetAssemblies be made properties of the implementation rather than the declaration?

**Comment by DmitrySikorsky** [comment]
Yes, you are right. I think we could have 1 method in the interface:

`GetAssemblies(IAssemblyProviderOptions options)`

(But in this case IAssemblyProviderOptions will be just marker interface with no properties or methods defined. What do you think about that?)

**Comment by fraserkyle** [comment]
Perfect.  This would leave the options open to extension.  :-)
